lib/auth/auth.go

```diff
@@ -4307,14 +4307,43 @@ func (a *Server) DeleteKubernetesCluster(ctx context.Context, name string) error
 	return nil
 }
 
-// SubmitUsageEvent submits an external usage event.
-func (a *Server) SubmitUsageEvent(ctx context.Context, req *proto.SubmitUsageEventRequest) error {
+// convertUsageEventUserMetadata is used to provide User Metadata queries to the ConvertUsageEvent.
+//
+// Fetching them ahead of time may cause some unnecessary queries, so the converted is
+// responsible for calling those methods as needed.
+type convertUsageEventUserMetadata struct {
+	*Server
```

Comment:
Does the `convertUsageEventUserMetadata` object needs to embed the whole `Server` implementing ? 

Based on implementation in needs only: 
```go
type userGetter interface {
 GetUser(name string, withSecrets bool) (types.User, error)
}
```

so 
```suggestion
	userGetter userGetter 
```
for me is more readable. 

Also the object name is convertUsageEventUserMetadata. How this object itself does convert logic  ?

Reply:
It doesn't :+1: 
However, we are only trying to satisfy the `usagereporter.UserMetadataClient` interface

Renamed the struct to `userMetadataGetter`